### ISSUE TITLE
creating COB from .csv

### ISSUE THREAD

**Issue by MeeshCompBio** [schae234/Camoco#26]
Hi @schae234,

I used to be able to build networks manually from .csv's until the latest git pull I made last week (I am still able to build networks with .tsv values). It seems to be an HD5/tables issue but I can't seem to figure out why this error would pop up for a .csv and not a .tsv . I made sure to delete any remnants of the network I was trying to build before attempting to build it again. 

I just wanted to double check that a .csv table would look something like the following as input and nothing has changed format wise;

```
LineA,LineB,LineC
GeneA,1,2,3
GeneB,2,2,2
```

.tsv build command that completes
```
co.COB.from_table('ROOTFPKM.tsv.gz',
'MeeshZmRoot',
'MeeshMaize Root Network',
Zm5bFGS,
rawtype='RNASEQ',
max_gene_missing_data=0.3,
max_accession_missing_data=0.08,
min_single_sample_expr=1,
min_expr=0.001,
quantile=False,
max_val=300
)

```

.csv file/command that used to work but now fails
```
In [8]: Zm5bFGS = co.RefGen('Zm5bFGS')

In [9]: co.COB.from_table('SubRoot.csv',
   ...: 'ZmSubRoot',
   ...: 'Maize SubRoot Network',
   ...: Zm5bFGS,
   ...: rawtype='RNASEQ',
   ...: max_gene_missing_data=0.3,
   ...: max_accession_missing_data=0.08,
   ...: min_single_sample_expr=1,
   ...: min_expr=0.001,
   ...: quantile=False,
   ...: sep=',',
   ...: max_val=300
   ...: )
[LOG] Mon Feb 15 06:54:30 2016 - Loading Expr table
[LOG] Mon Feb 15 06:54:31 2016 - Building Expr Index
[LOG] Mon Feb 15 06:54:31 2016 - Loading RefGen
[LOG] Mon Feb 15 06:54:31 2016 - RefGen for ZmSubRoot not set!
[LOG] Mon Feb 15 06:54:31 2016 - Loading coex table
[LOG] Mon Feb 15 06:54:31 2016 - ZmSubRoot is empty ('No object named coex in the file')
[LOG] Mon Feb 15 06:54:31 2016 - Loading Global Degree
[LOG] Mon Feb 15 06:54:31 2016 - ZmSubRoot is empty ('No object named degree in the file')
[LOG] Mon Feb 15 06:54:31 2016 - Resetting raw expression data
[LOG] Mon Feb 15 06:54:31 2016 - Resetting expression data
[LOG] Mon Feb 15 06:54:31 2016 - Extracting raw expression values
[LOG] Mon Feb 15 06:54:31 2016 - Importing Raw Expression Values
[LOG] Mon Feb 15 06:54:31 2016 - Trans. Log: raw->RawRNASEQ
[LOG] Mon Feb 15 06:54:31 2016 - Resetting expression data
[LOG] Mon Feb 15 06:54:31 2016 - Extracting raw expression values
[LOG] Mon Feb 15 06:54:31 2016 - Performing Quality Control on genes
[LOG] Mon Feb 15 06:54:31 2016 - ------------Quality Control
[LOG] Mon Feb 15 06:54:31 2016 - Raw Starting set: 39655 genes 26 accessions
[LOG] Mon Feb 15 06:54:31 2016 - Filtering out genes not in Reference Genome: Zea Mays - 5b - Zm5bFGS
[LOG] Mon Feb 15 06:54:32 2016 - Filtering out expression values lower than 0.001
[LOG] Mon Feb 15 06:54:32 2016 - Filtering out genes with > 0.3 missing data
[LOG] Mon Feb 15 06:54:37 2016 - Filtering out genes which do not have one sample above 1
[LOG] Mon Feb 15 06:54:51 2016 - Filtering out accessions with > 0.08 missing data
[LOG] Mon Feb 15 06:54:53 2016 - Gene Removal:
has_id                 39655
pass_membership        39655
pass_missing_data      25401
pass_min_expression    30663
PASS_ALL               25203
dtype: int64
[LOG] Mon Feb 15 06:54:53 2016 - Accession Removal:
has_id               26
pass_missing_data    26
PASS_ALL             26
dtype: int64
[LOG] Mon Feb 15 06:54:57 2016 - Genes passing QC by chromosome:
         has_id  pass_membership  pass_missing_data  pass_min_expression  PASS_ALL
chrom                                                                             
1          6056             6056               3929                 4690      3898
10         2727             2727               1731                 2113      1715
2          4766             4766               3002                 3674      2979
3          4197             4197               2743                 3317      2726
4          4197             4197               2592                 3159      2576
5          4503             4503               3037                 3575      3013
6          3293             3293               2133                 2574      2112
7          3147             3147               2028                 2460      2012
8          3531             3531               2341                 2820      2319
9          3006             3006               1865                 2281      1853
Mt          123              123                  0                    0         0
Pt           57               57                  0                    0         0
UNKNOWN      52               52                  0                    0         0
[LOG] Mon Feb 15 06:54:57 2016 - Kept: 25203 genes 26 accessions
[LOG] Mon Feb 15 06:54:57 2016 - Trans. Log: raw->quality_control
[LOG] Mon Feb 15 06:54:58 2016 - Performing Raw Expression Normalization
[LOG] Mon Feb 15 06:54:58 2016 - ------------ Normalizing
[LOG] Mon Feb 15 06:54:58 2016 - Trans. Log: raw->quality_control->arcsine
[LOG] Mon Feb 15 06:54:58 2016 - Filtering refgen: Zm5bFGS
[LOG] Mon Feb 15 06:55:00 2016 - Adding 25203 Genes info to database
[LOG] Mon Feb 15 06:55:00 2016 - Adding Gene attr info to database
[LOG] Mon Feb 15 06:55:01 2016 - Building Indices
[LOG] Mon Feb 15 06:55:46 2016 - Calculating Coexpression
[LOG] Mon Feb 15 06:59:03 2016 - Calculating Gene Distance
Calculating for 25203 genes
[LOG] Mon Feb 15 06:59:17 2016 - Indexing coex table
[LOG] Mon Feb 15 07:00:09 2016 - Building Database
[LOG] Mon Feb 15 07:00:21 2016 - Something bad happened:Problems creating the Array.
---------------------------------------------------------------------------
HDF5ExtError                              Traceback (most recent call last)
<ipython-input-9-f87b9b275298> in <module>()
     10 quantile=False,
     11 sep=',',
---> 12 max_val=300
     13 )

/project/chadm/Meesh/CamocoHelixInstall/Camoco/camoco/COB.py in from_table(cls, filename, name, description, refgen, rawtype, sep, index_col, **kwargs)
   1023             ),
   1024             name,description,refine,
-> 1025             rawtype=rawtype,**kwargs
   1026         )
   1027 

/project/chadm/Meesh/CamocoHelixInstall/Camoco/camoco/COB.py in from_DataFrame(cls, df, name, description, refgen, rawtype, **kwargs)
    972             df, name, description, refgen, rawtype, **kwargs
    973         )
--> 974         return cls.from_Expr(expo)
    975 
    976     @classmethod

/project/chadm/Meesh/CamocoHelixInstall/Camoco/camoco/COB.py in from_Expr(cls, expo)
    929         # The Expr object already exists, just get a handle on it
    930         self = expo
--> 931         self._calculate_coexpression()
    932         self._calculate_degree()
    933         self._calculate_leaves()

/project/chadm/Meesh/CamocoHelixInstall/Camoco/camoco/COB.py in _calculate_coexpression(self, significance_thresh)
    800 
    801         # 5. Put table in the hdf5 store
--> 802         self._build_tables(tel)
    803         self.log("Done")
    804         return self

/project/chadm/Meesh/CamocoHelixInstall/Camoco/camoco/COB.py in _build_tables(self, tel)
    861             self.log("Building Database")
    862             ## HDF5 Store
--> 863             self.hdf5['coex'] = tel
    864             self.log("Flushing Database")
    865             self.hdf5.flush(fsync=True)

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in __setitem__(self, key, value)
    419 
    420     def __setitem__(self, key, value):
--> 421         self.put(key, value)
    422 
    423     def __delitem__(self, key):

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in put(self, key, value, format, append, **kwargs)
    817             format = get_option("io.hdf.default_format") or 'fixed'
    818         kwargs = self._validate_format(format, kwargs)
--> 819         self._write_to_group(key, value, append=append, **kwargs)
    820 
    821     def remove(self, key, where=None, start=None, stop=None):

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in _write_to_group(self, key, value, format, index, append, complib, encoding, **kwargs)
   1266 
   1267         # write the object
-> 1268         s.write(obj=value, append=append, complib=complib, **kwargs)
   1269 
   1270         if s.is_table and index:

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in write(self, obj, **kwargs)
   2812             # I have no idea why, but writing values before items fixed #2299
   2813             blk_items = data.items.take(blk.mgr_locs)
-> 2814             self.write_array('block%d_values' % i, blk.values, items=blk_items)
   2815             self.write_index('block%d_items' % i, blk_items)
   2816 

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in write_array(self, key, value, items)
   2593                         self.group, key)._v_attrs.value_type = 'timedelta64'
   2594                 else:
-> 2595                     self._handle.create_array(self.group, key, value)
   2596 
   2597         getattr(self.group, key)._v_attrs.transposed = transposed

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/tables/file.py in create_array(self, where, name, obj, title, byteorder, createparents, atom, shape)
   1150         parentnode = self._get_or_create_path(where, create parents
   1151         return Array(parentnode, name,
-> 1152                      obj=obj, title=title, byteorder=byteorder)
   1153 
   1154     createArray = previous_api(create_array)

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/tables/array.py in __init__(self, parentnode, name, obj, title, byteorder, _log, _atom)
    185         # Ordinary arrays have no filters: leaf is created with default ones.
    186         super(Array, self).__init__(parentnode, name, new, Filters(),
--> 187                                     byteorder, _log)
    188 
    189     def _g_create(self):

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/tables/leaf.py in __init__(self, parentnode, name, new, filters, byteorder, _log)
    260         # is a lazy property that automatically handles their loading.
    261 
--> 262         super(Leaf, self).__init__(parentnode, name, _log)
    263 
    264     def __len__(self):

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/tables/node.py in __init__(self, parentnode, name, _log)
    265             #   Create or open the node and get its object ID.
    266             if new:
--> 267                 self._v_objectid = self._g_create()
    268             else:
    269                 self._v_objectid = self._g_open()

/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/tables/array.py in _g_create(self)
    217             # on
    218             (self._v_objectid, self.shape, self.atom) = self._create_array(
--> 219                 nparr, self._v_new_title, self.atom)
    220         except:  # XXX
    221             # Problems creating the Array on disk. Close node and re-raise.

tables/hdf5extension.pyx in tables.hdf5extension.Array._create_array (tables/hdf5extension.c:13312)()

HDF5ExtError: Problems creating the Array.

```

Thanks for the help!

**Comment by schae234** [comment]
Is the disk full?

**Comment by MeeshCompBio** [comment]
Yep, that explains it...

**Comment by schae234** [comment]
Making the files a little slimmer is on the TODO list...

Also, did you see that comment by the pandas people on line-arrow 2812 in  of the traceback?

```
/project/chadm/Meesh/CamocoHelixInstall/.conda/CMV/lib/python3.4/site-packages/pandas/io/pytables.py in write(self, obj, **kwargs)
   2812             # I have no idea why, but writing values before items fixed #2299
   2813             blk_items = data.items.take(blk.mgr_locs)
-> 2814             self.write_array('block%d_values' % i, blk.values, items=blk_items)
   2815             self.write_index('block%d_items' % i, blk_items)
   2816 
```
Good to know they struggle too!
